app/src/main/java/org/mozilla/focus/home/HomeTileScreenshotStore.kt

```diff
@@ -134,7 +136,7 @@ object HomeTileScreenshotStore {
             null
         } else {
             file.inputStream().use {
-                BitmapFactory.decodeStream(it, null, BITMAP_FACTORY_OPTIONS)
+                BitmapFactory.decodeStream(it, null, BITMAP_FACTORY_OPTIONS).withRoundedCorners(context.resources.getDimension(R.dimen.home_tile_corner_radius))
```

Comment:
Oh, I see what the problem is here: we need to round in two different places. I don't think we should be modifying the items in the store: let's modify them right before we set them (switching on the tile type) (i.e. don't round the corners inside the placeholderGenerator too). e.g.:
```kotlin
        val screenshotDeferred = async {
            val screenshot = HomeTileScreenshotStore.read(itemView.context, item.id)?.withRoundedCorners(...)
            screenshot ?: HomeTilePlaceholderGenerator.generate(itemView.context, item.url).withRoundedCorners(...)
        }
```

I prefer this approach because it keeps the related code together (and hopefully reduces duplication) and because the HomeTileScreenshotStore should have one responsibility: retrieving items from disk verbatim and that's it (e.g. if we wanted to create a cache, I'd wrap the Store we have rather than putting a cache inside of it).